### PR TITLE
systemvmtemplate: use latest LTS kernel from buster-ports

### PR DESCRIPTION
**Pull request by rhtyd**
Use the latest v5.10 Linux kernel (LTS) from buster-ports and fixes systemvmtemplate build issue on `main` branch.

Signed-off-by: Rohit Yadav <rohit.yadav@shapeblue.com>

### CODE REVIEW

**Comment by weizhouapache** [comment]
@rhtyd good.
maybe consider #3900 and set a smaller template size ?

**Comment by rhtyd** [comment]
@weizhouapache I think it would be too much maintenance work for gain to maintain separate systemvmtemplate for each hypervisor. These days users can download 100MBs more in few extra seconds.

File: tools/appliance/systemvmtemplate/scripts/apt_upgrade.sh [hunk 1]
```diff
@@ -53,11 +53,17 @@ function apt_upgrade() {
...
   apt-get -q -y upgrade
   apt-get -q -y dist-upgrade
+  apt-get -q -y upgrade -t buster-backports
+  apt-get -q -y dist-upgrade -t buster-backports
+
   apt-get -y autoremove --purge
   apt-get autoclean
   apt-get clean
+  reboot
 }
```
Comment (weizhouapache) [inline tools/appliance/systemvmtemplate/scripts/apt_upgrade.sh:67]: @rhtyd might 'reboot' be an issue in packer processing ?

File: tools/appliance/systemvmtemplate/scripts/configure_grub.sh [hunk 2]
```diff
@@ -20,6 +20,8 @@ set -e
 set -x
 
 function configure_grub() {
+  # Remove old kernel
+  apt-get remove -y --purge linux-image-amd64 linux-image-4.19.0-16-amd64 || true
```
Comment (weizhouapache) [inline tools/appliance/systemvmtemplate/scripts/configure_grub.sh:24]: @rhtyd two potential issues in my opinion
(1) the versions are different in debian releases (debian 10.9.0: kernel 4.19.0.16.debian 10.8.0: kernel 4.19.0.14,etc)
(2) might there be more than 2 versions ? for example, one new kernel by `apt-get -q -y dist-upgrade`, another new kernel by `apt-get -q -y dist-upgrade -t buster-backports` ?

**Comment by weizhouapache** [comment]
> @weizhouapache I think it would be too much maintenance work for gain to maintain separate systemvmtemplate for each hypervisor. These days users can download 100MBs more in few extra seconds.

@rhtyd yes, agree. let's use same template for hypervisors.

File: tools/appliance/systemvmtemplate/scripts/apt_upgrade.sh (hunk 1 above)
Comment (rhtyd) [reply tools/appliance/systemvmtemplate/scripts/apt_upgrade.sh:67]: No see template.json

File: tools/appliance/systemvmtemplate/scripts/configure_grub.sh (hunk 2 above)
Comment (rhtyd) [reply tools/appliance/systemvmtemplate/scripts/configure_grub.sh:24]: I see, we can explore some general way of removing kernels that are not in use.

File: tools/appliance/systemvmtemplate/template.json [hunk 3]
```diff
@@ -6,7 +6,14 @@
       "type": "shell",
       "execute_command": "echo 'cloud' | sudo -u root -S bash {{.Path}}",
       "scripts": [
-        "scripts/apt_upgrade.sh",
+        "scripts/apt_upgrade.sh"
+      ],
+      "expect_disconnect": true
```
Comment (rhtyd) [inline tools/appliance/systemvmtemplate/template.json:11]: @weizhouapache this handles for reboot case. You can read more about it here https://www.packer.io/docs/provisioners/shell

File: tools/appliance/systemvmtemplate/scripts/configure_grub.sh [hunk 4]
```diff
@@ -20,6 +20,8 @@ set -e
 set -x
 
 function configure_grub() {
+  # Remove old kernel
+  apt-get remove -y --purge linux-image-amd64 linux-image-4* || true
```
Comment (rhtyd) [inline tools/appliance/systemvmtemplate/scripts/configure_grub.sh:24]: @weizhouapache would this help, essentially saying remove all the old 4.x kernels. As the new/LTS Linux kernel are in the 5.x series - https://www.kernel.org/category/releases.html

**Comment by rhtyd** [comment]
Is this good to be merged @weizhouapache @Pearl1594

**Review by weizhouapache — APPROVED** [review]
@rhtyd yes, it is good to me.

Will we wait for debian 11 which is planned to be released in July ?

**Comment by rhtyd** [comment]
@weizhouapache yes we should update to Debian 11 if/when it comes before 4.16. In the meanwhile we can use the same Linux kernel backport same as Debian 11.

**Comment by rhtyd** [comment]
@blueorangutan package

File: tools/appliance/systemvmtemplate/scripts/install_systemvm_packages.sh [hunk 5]
```diff
@@ -22,7 +22,7 @@ set -x
 function install_vhd_util() {
   [[ -f /bin/vhd-util ]] && return
 
-  wget --no-check-certificate https://github.com/shapeblue/cloudstack-nonoss/raw/master/vhd-util -O /bin/vhd-util
+  wget --no-check-certificate https://github.com/shapeblue/cloudstack-nonoss/raw/main/vhd-util -O /bin/vhd-util
```
Comment (rhtyd) [inline tools/appliance/systemvmtemplate/scripts/install_systemvm_packages.sh:25]: jfyi @sureshanaparti - fixed this

Comment (sureshanaparti) [reply tools/appliance/systemvmtemplate/scripts/install_systemvm_packages.sh:25]: noted, thanks @rhtyd

**Comment by rhtyd** [comment]
@blueorangutan test matrix

**Comment by rhtyd** [comment]
@blueorangutan test matrix

**Comment by weizhouapache** [comment]
@rhtyd good. are you going to build new systemvm template and upload to download.cloudstack.org ?

**Comment by rhtyd** [comment]
@weizhouapache the current 4.16 systemvmtemplate from this PR has been already on download.cloudstack.org for weeks now, all PRs being tested against that systemvmtemplate

**Comment by weizhouapache** [comment]
> @weizhouapache the current 4.16 systemvmtemplate from this PR has been already on download.cloudstack.org for weeks now, all PRs being tested against that systemvmtemplate

@rhtyd nice, good to know. thanks

**Pull request merged**